src/modules/control_allocator/ControlAllocator.cpp

```diff
@@ -335,13 +295,12 @@ ControlAllocator::Run()
 		_control_allocation->allocate();
 
 		// Publish actuator setpoint and allocator status
-		publish_actuator_setpoint();
-		publish_control_allocator_status();
+		publish_actuator_controls();
 
-		// Publish on legacy topics for compatibility with
-		// the current mixer system and multicopter controller
-		// TODO: remove
-		publish_legacy_actuator_controls();
+		if (now - _last_status_pub >= 5_ms) {
```

Comment:
Not that important, but why rate limit it here at 5 ms?

You could limit it by only publishing when there's a non-negligible change in allocated control (or something similar).

Reply:
It turns out it's quite expensive at high rate (https://github.com/PX4/PX4-Autopilot/pull/18563/commits/a215bfd6319bfe5aa5cabda6e05ea64b43ca5fbd), and as it's used for slower dynamics (i.e. integrator anti-windup), I generally limited the rate.
I can add a comment.